classes/i18n/Locale.php

```diff
@@ -143,11 +144,8 @@ public function setLocale($locale): void
         }
 
         $this->locale = $locale;
-        $dashedLocale = str_replace('_', '-', $locale);
-        if (!setlocale(LC_ALL, "{$locale}.utf8", $locale, "{$dashedLocale}.utf8", $dashedLocale)) {
-            error_log("Failed to set locale \"{$locale}.utf8\", ensure that all locales enabled in the application are available in your operational system");
-        }
-        putenv("LC_ALL={$locale}");
+        setlocale(LC_ALL, 'C.utf8', 'C');
```

Comment:
@jonasraoni, do you know what the effect of `setlocale` is when `Locale::setDefault` is also used? If we're just going to force `C` or `C.utf8`, why not just leave out the `setlocale` call entirely?

Reply:
do you know what the effect of setlocale is when Locale::setDefault is also used? >

@asmecher The `Locale::setDefault()` is applied just to the [intl functions](https://www.php.net/manual/en/book.intl.php), it doesn't affect the `strftime()` for example.

> why not just leave out the setlocale call entirely?

I've left just because of this warning: https://www.php.net/setlocale#:~:text=Warning,of%20PHP%207.0.5.
The use case would be a server, like Apache, holding PHP under a thread, and one of them calls the `setlocale()` (perhaps another application like WordPress)... Basically a race-condition 😁

I'd leave it this way, but if you prefer to drop, no objections. I'll wait for a direction 😁

Reply:
OK, but in the previous behaviour, functions like `strftime` would be localized properly via the user of `setlocale` with the appropriate `$locale` option (assuming that's available on the server, which ideally is the case). Now these functions will behave with a hard-coded `C` locale. We're not using `strftime` anymore (deprecated), but off the top of my head, this would still include e.g. number formatting etc.

Reply:
Yeah! So, what I'm proposing is to use the `C`, and fix any problem that might arise (if there's something bad, it went unnoticed for all these years, so I'm not very scared about it) through any other way (the `intl` classes must cover probably all the cases).

Reply:
Another option is to map our locale to a better one (using the `ResourceBundle::getLocales('')`) in `3.4`, and drop the `setlocale()` just in `3.5`.

Reply:
I don't think I see much value in making any changes to stable-3_3_0, if we don't have any reports of problems. In other words, what's the harm in reverting https://github.com/pkp/pkp-lib/commit/c80e90ca19f209f3deae2cb1d813d4f34daa8966 and leaving stable-3_3_0 as is?

(I don't consider stable-3_4_0 to be quite the same case -- setlocale doesn't expect RFC 4646 locale names, which we've adopted more broadly starting with 3.4.0-0.)

Reply:
About `stable-3_3_0`, if I have a more predictable option available, that's the one I take, but I don't have anything against reverting.
I'll wait for a final direction, my preferences are expressed on the PRs :)